flocx_market/db/sqlalchemy/api.py

```diff
@@ -89,7 +95,7 @@ def offer_update(marketplace_offer_id, values, context):
         offer_ref.save(get_session())
         return offer_ref
     else:
-        return None
+        raise exception.ResourceNotFound(resource_type="Offer")
```

Comment:
Is this going to work?  The `ResourceNotFound` class has `msg_fmt = "{resource_type} {resource_uuid} not found."`, so this will just hit the `KeyError` in [line 12](https://github.com/CCI-MOC/flocx-market/pull/109/files#diff-a108e762755a42368a75f9314b728caeR12).

Reply:
It actually does not break for reasons I don't understand. Instead it returns
{resource_type} {resource_uuid} not found.
Strangely, resource_type does not get parsed, even though I did pass in that argument. I am trying to find a more elegant solution. This is initially why I had a third error class

Reply:
Passing in kwargs instead of **kwargs causes a Keyerror however

Reply:
It returns `{resource_type} {resource_uuid} not found` because it generates ` KeyError`, since the format string expects both `resource_type` and `resource_uuid` but you're only providing `resource_type`.  Since there is no `resource_uuid`, you get a `KeyError`, and it defaults to using the raw `msg_format` instead.

Reply:
I figured it was some kind of error handling on behalf of format. Though why **kwargs works but not kwargs is strange. Has to do with error handling on unpacked items within the format function. Regardless, I am going to just make a third error class. This error is actually not even needed here. Mainn said get_all should just return an empty list when the queries find nothing

Reply:
But in the contract code this third class will be needed